src/java/org/apache/cassandra/dht/RangeStreamer.java

```diff
@@ -358,12 +358,7 @@ public StreamResultFuture fetchAsync()
             InetAddress preferred = SystemKeyspace.getPreferredIP(source);
             Collection<Range<Token>> ranges = entry.getValue().getValue();
 
-            // filter out already streamed ranges
-            Set<Range<Token>> availableRanges = stateStore.getAvailableRanges(keyspace, StorageService.getPartitioner());
-            if (ranges.removeAll(availableRanges))
-            {
-                logger.info("Some ranges of {} are already available. Skipping streaming those ranges.", availableRanges);
-            }
+            removeAvailableRanges(entry, true);
```

Comment:
Doesn't this mean that we aren't actually removing these ranges from `ranges` anymore?